packages/design-system/src/components/Form/Primitives/Field/Field.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { forwardRef, ReactElement, Ref, cloneElement } from 'react';
+import Link, { LinkProps } from '../../../Link/Link';
+import { StackVertical } from '../../../Stack';
+import Label, { LabelProps } from '../Label/Label';
+import { InlineMessageDestructive, InlineMessageInformation } from '../../../InlineMessage';
+import VisuallyHidden from '../../../VisuallyHidden';
+
+export type FieldStatusProps = {
+	isError?: boolean;
+};
+
+type FieldProps = {
+	link?: LinkProps;
+	description?: string;
+	hideLabel?: boolean;
+	label: LabelProps;
+	id: string;
+	name: string;
+} & FieldStatusProps;
+
+type FieldPropsWithChildren = FieldProps & { children: ReactElement };
+
+const Field = forwardRef(
+	(props: FieldPropsWithChildren, ref: Ref<HTMLInputElement | HTMLTextAreaElement>) => {
+		const {
+			children,
+			link,
+			id,
+			label,
+			name,
+			hideLabel = false,
+			isError = false,
+			description,
+			...rest
+		} = props;
+
+		const LabelComponent = hideLabel ? (
+			<VisuallyHidden>
+				<Label {...label} htmlFor={id} />
+			</VisuallyHidden>
+		) : (
+			<Label {...label} htmlFor={id} />
+		);
+
+		const Description = ({ descriptionText }: { descriptionText: string }) => {
+			const descProps = {
+				description: descriptionText,
+			};
+			if (isError) {
+				return <InlineMessageDestructive {...descProps} />;
+			}
+			return <InlineMessageInformation {...descProps} />;
+		};
```

Comment:
I replicated the current Field component's API but I did ask myself the same question. 
My reasoning in keeping it was that I don't like having two `string` props but no way to display both values.

Reply:
I see it as 2 possible strings, and a switch `has/isError` to select the one we want.
I won't block the PR for that, it's a choice to make, and if you think the api is better like that, that's not a big deal 🤷‍♂️

Reply:
Feel free to resolve the conversation if you decide to keep it as is

Reply:
I'm currently trying the implem to see how it feels. So far it feels like extra work for everybody involved :D

Reply:
I've opted for a single `description` props, sorry. But thanks for the suggestion though because it made me rework the typing. Now `description` is mandatory in `hasError` cases. I've updated the story to showcase that.